src/Microsoft.DotNet.Build.Tasks.net45/ExtractResWResourcesFromAssemblies.cs

```diff
@@ -0,0 +1,261 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections;
+using System.IO;
+using System.Reflection;
+using System.Resources;
+using System.Xml.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class ExtractResWResourcesFromAssemblies : Task
+    {
+        [Required]
+        public ITaskItem[] InputAssemblies { get; set; }
+
+        [Required]
+        public string OutputPath { get; set; }
+
+        [Required]
+        public string InternalReswDirectory { get; set; }
+
+        public override bool Execute()
+        {
+            try
+            {
+                Directory.CreateDirectory(OutputPath);
+                CreateReswFiles();
+            }
+            catch (Exception e)
+            {
+                Log.LogErrorFromException(e, showStackTrace: false);
+            }
+
+            return !Log.HasLoggedErrors;
+        }
+
+        public void CreateReswFiles()
+        {
+            foreach (ITaskItem assemblySpec in InputAssemblies)
+            {
+                string assemblyPath = assemblySpec.ItemSpec;
+
+                if (!ShouldExtractResources($"FxResources.{Path.GetFileNameWithoutExtension(assemblyPath)}.SR.resw", assemblyPath))
+                {
+                    continue; // we skip framework assemblies that resources already exist and don't need to be extracted to avoid reading dll metadata.
+                }
+                
+                try
+                {
+                    Assembly assembly = Assembly.LoadFrom(assemblyPath);
```

Comment:
Are we every going to hit an assembly load failure due to missing dependencies? For example if we load a .NET Core assembly on .NET Framework then it might not have all the dependencies resolved.